src/activity_handlers.cpp

```diff
@@ -481,417 +481,195 @@ int butcher_time_to_cut( const player &u, const item &corpse_item, const butcher
+        } else if( mt.has_flag( MF_HUMAN ) ) {
+            harvest_id_name = "human";
+        } else if( mt.size >= MS_SMALL && mt.size <= MS_MEDIUM ) {
+            if( mt.has_flag( MF_LEATHER ) ) {
+                harvest_id_name = "mammal_leather";
+            } else if( mt.has_flag( MF_FUR ) ) {
+                harvest_id_name = "mammal_fur";
+            } else if( mt.has_flag( MF_WOOL ) ) {
+                harvest_id_name = "mammal_wool";
+            } else if( mt.has_flag( MF_FEATHER ) ) {
+                harvest_id_name = "bird_small";
+            } else if( mt.has_flag( MF_AQUATIC ) ) {
+                harvest_id_name = "fish_large";
+            } else if( mt.has_flag( MF_CHITIN ) ) {
+                harvest_id_name = "arachnid";
+            } else if( mt.has_flag( MF_BONES ) ) {
+                harvest_id_name = "animal_noskin";
+            } else {
+                harvest_id_name = "meatslug";
+            }
+        } else if( mt.size >= MS_LARGE ) {
+            if( mt.has_flag( MF_LEATHER ) ) {
+                harvest_id_name = "mammal_large_leather";
+            } else if( mt.has_flag( MF_FUR ) ) {
+                harvest_id_name = "mammal_large_fur";
+            } else if( mt.has_flag( MF_WOOL ) ) {
+                harvest_id_name = "mammal_large_wool";
+            } else if( mt.has_flag( MF_FEATHER ) ) {
+                harvest_id_name = "bird_large";
+            } else if( mt.has_flag( MF_AQUATIC ) ) {
+                harvest_id_name = "fish_large";
+            } else if( mt.has_flag( MF_CHITIN ) ) {
+                harvest_id_name = "arachnid";
+            } else if( mt.has_flag( MF_BONES ) ) {
+                harvest_id_name = "animal_large_noskin";
+            } else {
+                harvest_id_name = "meatslug";
             }
         }
     }
 
-    if( ( corpse->has_flag( MF_FUR ) || corpse->has_flag( MF_LEATHER ) ||
-          corpse->has_flag( MF_CHITIN ) ) && skins > 0 ) {
-        p->add_msg_if_player( m_good, _( "You manage to skin the %s!" ), corpse->nname().c_str() );
-        int fur = 0;
-        int tainted_fur = 0;
-        int leather = 0;
-        int tainted_leather = 0;
-        int human_leather = 0;
-        int chitin = 0;
+    return harvest_list::all().at( harvest_id( harvest_id_name ) );
+}
+
+// this function modifies the input weight by its damage level, depending on the bodypart
+int corpse_damage_effect( int weight, std::string entry_type, int damage_level )
+{
+    float slight_damage = 0.9;
+    float damage = 0.75;
+    float high_damage = 0.5;
+    int destroyed = 0;
 
-        while( skins > 0 ) {
-            if( corpse->has_flag( MF_CHITIN ) ) {
-                chitin = rng( 0, skins );
-                skins -= chitin;
-                skins = std::max( skins, 0 );
+    switch( damage_level ) {
+        case 2: // "damaged"
+            if( entry_type == "offal" ) {
+                return round( weight * damage );
```

Comment:
`round` is used very often, but the input value is usually a number of grams. I suggest changing this to use `units::weight` consistently for all weights and getting rid of the rounding of weight values. This will loose up to a single gram, which seems *very* insignificant.

It would make the code much simpler and easier to follow.

Reply:
this will cause a lot of unit changes back and forth in order to use `rng()`. however if i do not convert `roll` to a `units::mass` then very little will be changed here in any case.

Reply:
There is an implementation of `rng` with `units::quantity` in "artifact.cpp", you may have to move it into some header to access it here.

```C++
rng( 0_gram, 1000_gram ); // will also return units::weight
```

I don't see that many invocations of `rng` with a weight anyway (in fact just one, the other calls to `rng` use counts of items and similar).

Reply:
`roll` actually pulls double duty here, and i thought you meant for me to also make that into `units::mass` but if `monster_weight` and `monster_weight_remaining` are the only ones you suggest to change, then that's a fairly minor and easy fix

Reply:
i took another look at the code, and `round()` is being used 1 time in the section of the code you suggested i change to `units::mass`. i suppose i can change it still, but the other two instances of `round()` are not in a spot that would lend well to using `units::mass`